ocaml/xapi/xapi_vm_snapshot.ml

```diff
@@ -409,14 +409,28 @@ let extended_do_not_copy =
 
 let domain_type_to_string = function `pv -> "pv" | `hvm -> "hvm"
 
-(* Update the domain_type field if it does not match the HVM_boot_policy *)
+(* Update the domain_type field if it does not match the HVM boot policy *)
 let ensure_domain_type_is_consistent ~__context ~snap_metadata =
+  let hvm_boot_policy () =
+    (* XSI-828 the HVM boot policy is actually stored in the database as HVM__boot_policy
+     * we keep the check for HVM_boot_policy 'just in case' *)
+    let rec go = function
+      | [] ->
+          D.info "couldn't find HVM boot policy in snap metadata, assuming '%s'"
+            Constants.hvm_default_boot_policy ;
+          Constants.hvm_default_boot_policy
```

Comment:
You can't assume this. It is the same as saying that "I don't know the domain type, so I'll assume it's HVM".  If we can't find the domain_type _and_ not even the HVM_boot_policy, which is a highly unlikely (now that you've fixed it :) ), then I think it is better to make it a hard failure.